### PR TITLE
fix: Align and format column stats (new PR)

### PR DESCRIPTION
**Pull request by dikshathakur3119**
Signed-off-by: dikshathakur3119 <dikshathakur3119@gmail.com>

### Summary of Changes

Each column may have different column statistics like mean, median, etc. In this PR:

1. Added Numerical separator in the values using [Intl.NumberFormat().format(number)], this format can be configured, default is US format (e.g 1,234,567).

2. Right aligning numbers

3. Monospace font for the numbers

### Tests

Added test to see if configured number format is used to format column stats.
### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -70,7 +70,7 @@ responses==0.12.0
 
 # A common package that holds the models deifnition and schemas that are used
 # accross different amundsen repositories.
-amundsen-common>=0.3.6,<1.0
+amundsen-common==0.5.6,<1.0
```
Comment (feng-tao) [inline requirements.txt:73]: amundsen-common==0.5.6 is good enough

**Review by feng-tao — COMMENTED** [review]
+1 based on previous @Golodhros 's review

File: amundsen_application/static/css/_variables-default.scss [hunk 2]
```diff
@@ -128,6 +128,7 @@ $resource-title-color: $indigo60;
 
 // Spacing
 $spacer-size: 8px;
+$spacer-half: $spacer-size/2;
```
Comment (feng-tao) [inline amundsen_application/static/css/_variables-default.scss:131]: is $spacer-size/2 the convention or $spacer-size / 2?

Comment (Golodhros) [reply amundsen_application/static/css/_variables-default.scss:131]: I would say whatever stylelint does, if it doesn't care, we neither!

**Review by Golodhros — APPROVED** [review]
LGTM

**Review by Golodhros — APPROVED** [review]
LGTM

**Review by allisonsuarez — APPROVED** [review]

**Pull request merged**